listenbrainz/db/lb_radio_artist.py

```diff
@@ -0,0 +1,151 @@
+from collections import defaultdict
+import json
+from random import randint
+from typing import List
+
+from flask import current_app
+import psycopg2
+from psycopg2.extras import DictCursor
+
+from listenbrainz.webserver import ts_conn
+
+
+def lb_radio_artist(mode: str, seed_artist: str, max_similar_artists: int, num_recordings_per_artist: int, pop_begin: float,
+                    pop_end: float) -> List[dict]:
+    """
+        Fetch recordings for LB Radio's similar artists element.
+
+        Given a seed artist mbid, find similar artists given other parameters and then return
+        a dict of artist_mbids that contain lists of dict as such:
+            {
+              "recording_mbid": "401c1a5d-56e7-434d-b07e-a14d4e7eb83c",
+              "similar_artist_mbid": "cb67438a-7f50-4f2b-a6f1-2bb2729fd538",
+              "total_listen_count": 232361
+            }
+
+        Troi will take this data and complete processing it into a complete playlist.
+
+        parameters:
+
+        mode: LB radio mode, must be one of: easy, medium, hard.
+        seed_artist: artist mbid of the seed artist for similar artists
+        num_recordings_per_artist: Return up to this many recordings for each artist.
+        pop_begin: Popularity range percentage lower bound. A popularity range is given to narrow down
+                   the recordings into a smaller target group. The most popular track(s) on
+                   LB have a pop percent of 100. The least popular tracks have a score of 0.
+        pop_end: Popularity range percentage upper bound. See above.
+    """
+
+    query = """WITH mbids(mbid, score) AS (
```

Comment:
It would be best to pass `UUID`s to this query and then just keep using `UUID`s till the end and in the last select cast it to `TEXT`. This will ensure that we don't miss index use because of TEXT to UUID or vice versa conversion at some step.

Reply:
That sounds like a good policy. will do.